framework/include/vectorpostprocessors/LineMaterialSamplerBase.h

```diff
@@ -179,6 +184,7 @@ LineMaterialSamplerBase<T>::execute()
       addSample(_q_point[qp], qp_proj_dist_along_line, values);
     }
   }
+  _fe_problem.clearActiveMaterialProperties(_tid);
```

Comment:
We can always use RAII, but as you pointed out he's just following a pattern here. All of this code/logic looks like the variable dependency stuff. It's not really a hold up for this PR. RAII can have runtime costs so we have to consider those tradeoffs.